engine/execution/ingestion/uploader/retryable_uploader_wrapper.go

```diff
@@ -237,19 +237,46 @@ func (b *BadgerRetryableUploaderWrapper) reconstructComputationResult(
 		log.Warn().Msgf("failed to retrieve StateCommitment with BlockID %s. Error: %s", blockID.String(), err.Error())
 	}
 
+	executableBlock := &entity.ExecutableBlock{
+		Block:               block,
+		CompleteCollections: completeCollections,
+	}
+
+	compRes := execution.NewEmptyComputationResult(executableBlock)
+
+	eventsByTxIndex := make(map[int]flow.EventsList, 0)
+	for _, event := range events {
+		idx := int(event.TransactionIndex)
+		list, found := eventsByTxIndex[idx]
+		if found {
+			list = append(list, event)
+			continue
+		}
+		eventsByTxIndex[idx] = flow.EventsList{event}
+	}
+
+	lastChunk := len(completeCollections)
+	lastCollection := compRes.CollectionExecutionResultAt(lastChunk)
+	for i, txRes := range transactionResults {
+		lastCollection.AppendTransactionResults(
+			eventsByTxIndex[i],
+			nil,
+			nil,
+			txRes,
+		)
+	}
+
+	compRes.AppendCollectionAttestationResult(
+		endState,
```

Comment:
is this correct?

Reply:
I wonder why an uploader needs to know so many protocol details. 

I think the data conversion should be in the protocol layer, and the uploader should simply take an assembled data structure, encode it and upload to the bucket.

Reply:
its not 100% semantically accurate but it replicate what it was there before, populate fields that are only needed for forming updates, since we are going to depreciate uploader soon, I made it just work for now.